### PR TITLE
More intelligent Respect word detection

### PR DESCRIPTION
**Pull request by bimalkant-lauhny**
PR for less aggressive Respect Words detection. Currently bot detects words in unwanted places for ex, `sir` in `desires`! etc.. This PR fixes that!

### CODE REVIEW

File: observers/respect-terms.js [hunk 1]
```diff
@@ -8,13 +8,14 @@ var logging = require('../config/logger');
 
 module.exports = function(bot, from, to, message) {
 	var noRespectTerms = constants.terms.RESPECT;
```
Comment (paramsingh) [inline observers/respect-terms.js]: Maybe we should fix indentation here while we're at it?

**Review by paramsingh — CHANGES_REQUESTED** [review]
Looks good to me, just the one nitpick.

File: observers/respect-terms.js (hunk 1 above)
Comment (bimalkant-lauhny) [reply observers/respect-terms.js]: @paramsingh Done!

**Review by paramsingh — APPROVED** [review]
Woo :shipit:

**Pull request merged**